zone/bot.cpp

```diff
@@ -2469,13 +2468,19 @@ void Bot::AI_Process()
 	}
 
 	// We also need a leash owner and follow mob (subset of primary AI criteria)
-	Client* leash_owner = (bot_group->GetLeader() && bot_group->GetLeader()->IsClient() ? bot_group->GetLeader()->CastToClient() : bot_owner);
+	auto group_leader = bot_group->GetLeader();
+	Client* leash_owner = nullptr;
+	if (group_leader && group_leader->IsClient()) {
+		leash_owner = group_leader->CastToClient();
+	}
+	else {
+		leash_owner = bot_owner;
+	}
+
 	if (!leash_owner) {
```

Comment:
How does this end up being any different than what's happening in the ternary?

Reply:
dumb moment on my part! need to look into this further to determine where a poisoned ptr is coming from here.